### PR TITLE
Fixes line disabling flags from disabling max-line-length also on the following line (#1217)

### PR DESCRIPTION
**Pull request by vitaminj**
Fixes https://github.com/palantir/tslint/issues/1217

### CODE REVIEW

**Comment by JKillian** [comment]
@patsissons, since you worked on this recently, do these changes look good to you?

**Comment by patsissons** [comment]
My only suggestion would be to make sure that nested rules are not affected, I believe that's why I initially added the `+1` in there. But i had added that early on in development and so it may not be relevant any longer.

**Comment by vitaminj** [comment]
Yes if you have suggestions for other tests to add? I added an end-of-line one to vaguely test off-by-one errors at that end, but couldn't immediately think of anything to test start-of-line?

File: src/enableDisableRules.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ export class EnableDisableRulesWalker extends SkippableTokenAwareRuleWalker {
                         // end at the beginning of the next line
                         this.enableDisableRuleMap[ruleToAdd].push({
                             isEnabled: !isEnabled,
-                            position: scanner.getTextPos() + 1,
+                            position: scanner.getTextPos(),
```
Comment (JKillian) [inline src/enableDisableRules.ts]: Just thinking, maybe this would be a better way to get the beginning of the next line?
`position: this.getStartOfLinePosition(node, startingPosition, 1) - 1,`

**Comment by patsissons** [comment]
This is the best example test i can come up with off the top of my head, should validate both my original case and your new case:

```
// tslint:disable
// tslint:enable-next-line:max-line-length
var a = "testtesttesttesttesttesttesttesttesttesttesttesttest";
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ [exceeds maximum line length of 60]
var a = "testtesttesttesttesttesttesttesttesttesttesttesttest";
// tslint:enable
var a = "testtesttesttesttesttesttesttesttesttesttesttesttest";
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ [exceeds maximum line length of 60]

// tslint:disable-next-line:max-line-length
var a = "testtesttesttesttesttesttesttesttesttesttesttesttest";
var a = "testtesttesttesttesttesttesttesttesttesttesttesttest";
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ [exceeds maximum line length of 60]
```

**Comment by JKillian** [comment]
@vitaminj Let me know what you think of the above comments. If everything seems as good as possible to you, we can merge this as well

**Comment by patsissons** [comment]
i believe the test I added above will fail for one case with the current code, and fail for the other case in the proposed code (at least it did when i checked a few days ago).

**Comment by JKillian** [comment]
Gotchya, thanks for the heads-up @patsissons. Let's make sure that @patsissons's test case is handled correctly with the new code before we merge then

**Comment by adidahiya** [comment]
ping @vitaminj?
